src/test/scala/com/rawlabs/das/server/cache/iterator/QualSelectivityAnalyzerSpec.scala

```diff
@@ -0,0 +1,175 @@
+        // y is not mentioned => doesn't guarantee y=7
+      )
+
+      val result = QualSelectivityAnalyzer.differenceIfMoreSelective(oldQuals, newQuals)
+      result shouldBe None
+    }
+
+    it("should handle IsAnyQual (ANY => ANY) set logic") {
+      // old: color in {1,2,3}
+      // new: color in {1,2} => that is narrower => old => new?
+      // Actually, if color is in {1,2,3}, is that guaranteed to be in {1,2}? No, it might be 3 => new not satisfied.
+      // So old => new is false. That means new is more restrictive, so that can't be guaranteed => None if we try the reverse.
+      //
+      // Let's test the direction "a => b" with the sets. We do a direct check in a smaller test.
+      val oldQuals = Seq(isAnyQual("color", Operator.EQUALS, 1, 2, 3))
+      val newQuals = Seq(isAnyQual("color", Operator.EQUALS, 1, 2))
+
+      // We want to see if new is more selective. That means old => new must be true.
+      // But if color=3, old is satisfied, new is not => so old => new is false => return None
+      val res = QualSelectivityAnalyzer.differenceIfMoreSelective(oldQuals, newQuals)
+      res shouldBe None
+    }
+
+    it("should handle IsAllQual (ALL => ALL) set logic") {
+      // old: x >= ALL(10,15)
+      // new: x >= ALL(10,15,20)
+      //   old => new is false, because if x=17, it satisfies old but not new (fails x>=20).
+      //   So new is more restrictive => we can't guarantee old => new => None
+      // But we want new to be at least as selective => for new to cover old, we need old => new.
+      // Actually let's test the scenario the other way:
+      val oldQuals = Seq(isAllQual("x", Operator.GREATER_THAN_OR_EQUAL, 10, 15))
+      val newQuals = Seq(isAllQual("x", Operator.GREATER_THAN_OR_EQUAL, 10, 15, 20))
+      // If x satisfies old (x >=10 and x>=15 => x>=15), that doesn't guarantee x>=20.
+      // So no coverage => None
+      val res = QualSelectivityAnalyzer.differenceIfMoreSelective(oldQuals, newQuals)
+      res shouldBe None
+    }
+
+    it("should handle IsAnyQual subset properly for a => b case where bSet is subset of aSet") {
+      // old => color in {1,2,3,4}, new => color in {2,3}
+      // For old => new, we are checking if new is "covered"? Actually in differenceIfMoreSelective,
+      // we want each old to be covered by new => we do isCoveredBy(oldQ, newQuals), i.e. newQuals must have a qual that implies oldQ.
+      // We also do the difference by isAlreadyCovered(newQ, oldQuals).
+      //
+      // Actually let's invert the usual logic: If a => b to hold, we want b's set to be subset of a's set for ANY.
+      // But we have old=ANY(1,2,3,4) => new=ANY(2,3). This means if a value is in {1,2,3,4}, it's not necessarily in {2,3}. So a => b is false.
+      // But for the differenceIfMoreSelective, we want new => old or old => new? The code does old => new. So we expect None again.
+      val oldQuals = Seq(isAnyQual("color", Operator.EQUALS, 1, 2, 3, 4))
```

Comment:
This test looks quite like the other AnyQual test, they're feeding similar quals and assert the same. I've read the comment many times without understanding what it means.